cmd/argocd/commands/plugin.go

```diff
@@ -0,0 +1,142 @@
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// DefaultPluginHandler implements the PluginHandler interface
+type DefaultPluginHandler struct {
+	ValidPrefixes []string
+	lookPath      func(file string) (string, error)
+	run           func(cmd *exec.Cmd) error
+}
+
+// NewDefaultPluginHandler instantiates the DefaultPluginHandler
+func NewDefaultPluginHandler(validPrefixes []string) *DefaultPluginHandler {
+	return &DefaultPluginHandler{
+		ValidPrefixes: validPrefixes,
+		lookPath:      exec.LookPath,
+		run: func(cmd *exec.Cmd) error {
+			return cmd.Run()
+		},
+	}
+}
+
+// HandleCommandExecutionError processes the error returned from executing the command.
+// It handles both standard Argo CD commands and plugin commands. We don't require to return
+// error but we are doing it to cover various test scenarios.
+func (h *DefaultPluginHandler) HandleCommandExecutionError(err error, isArgocdCLI bool, args []string) error {
+	// the log level needs to be setup manually here since the initConfig()
+	// set by the cobra.OnInitialize() was never executed because cmd.Execute()
+	// gave us a non-nil error.
+	initConfig()
+	log.SetLevel(log.DebugLevel)
```

Comment:
Why setting debug logs? it should respect the `cmdutil.LogLevel` provided in the command arguments.